pkg/ssl.go

```diff
@@ -0,0 +1,204 @@
+package pkg
+
+import (
+	"encoding/pem"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"github.com/wttech/aemc/pkg/common/cryptox"
+	"github.com/wttech/aemc/pkg/common/filex"
+	"github.com/wttech/aemc/pkg/common/osx"
+	"github.com/wttech/aemc/pkg/common/pathx"
+	"golang.org/x/net/html"
+	"io"
+	"os"
+	"strings"
+)
+
+const (
+	SSLSetupPath = "/libs/granite/security/post/sslSetup.html"
+)
+
+type SSL struct {
+	instance *Instance
+}
+
+type sslLock struct {
+	KeystorePassword   string `yaml:"keystore_password"`
+	TrustStorePassword string `yaml:"trust_store_password"`
+	Certificate        string `yaml:"certificate"`
+	PrivateKey         string `yaml:"private_key"`
+	HttpsHostname      string `yaml:"https_hostname"`
+	HttpsPort          string `yaml:"https_port"`
+}
+
+func NewSSL(instance *Instance) *SSL {
+	return &SSL{
+		instance: instance,
+	}
+}
+
+func (s SSL) Setup(keyStorePassword, trustStorePassword, certificateFile, privateKeyFile, httpsHostname, httpsPort string) (bool, error) {
+	if !pathx.Exists(certificateFile) {
+		return false, fmt.Errorf("%s > certificate file does not exist: %s", s.instance.ID(), certificateFile)
+	}
+	if !pathx.Exists(privateKeyFile) {
+		return false, fmt.Errorf("%s > private key file does not exist: %s", s.instance.ID(), privateKeyFile)
+	}
+
+	privateKeyData, err := os.ReadFile(privateKeyFile)
+	if err != nil {
+		return false, fmt.Errorf("%s > failed to read private key file: %w", s.instance.ID(), err)
+	}
+	pemBlock, _ := pem.Decode(privateKeyData)
+	if pemBlock != nil {
+		tempDerFile, err := os.CreateTemp("", "aemc-private-key-*.der")
```

Comment:
may it have a chance to collide when run twice for local author then for local publish? how about consequences?

Reply:
https://pkg.go.dev/os#CreateTemp
Multiple programs or goroutines calling CreateTemp simultaneously will not choose the same file.